cmd/logger/reqinfo.go

```diff
@@ -120,3 +120,15 @@ func GetReqInfo(ctx context.Context) *ReqInfo {
 	}
 	return nil
 }
+
+// AddTagToContext adds a key-value tag to the ReqInfo to the given
+// context and returns the updated context. If the context does not
+// have a ReqInfo, it is also added.
+func AddTagToContext(ctx context.Context, key, value string) context.Context {
+	ri := GetReqInfo(ctx)
+	if ri == nil {
+		ri = &ReqInfo{}
+	}
+	ri.AppendTags(key, value)
+	return SetReqInfo(ctx, ri)
+}
```

Comment:
`AddTagToContext` is not going well with other relevant functions exposed by the logger package like SetReqInfo and GetReqInfo. Tag related methods are only on ReqInfo struct.

The need for `AddTagToContext` seems to arise from the situation where ctx is not passed to a given function (and hence we have to do context.Background), or even if it is passed we dont have a valid ReqInfo (because someone passed us just context.Background() which does not have any ReqInfo)

Instead of adding a new API, we can do this:
```
// GetReqInfo returns ReqInfo if set.
func GetReqInfo(ctx context.Context) *ReqInfo {
	if ctx != nil {
		r, ok := ctx.Value(contextLogKey).(*ReqInfo)
		if ok {
			return r
		}
		r = &ReqInfo{}
		SetReqInfo(ctx, r)
		return r
	}
	return nil
}
```

The advantage is there is we don't have to know when to use AddTagToContext and instead use GetInfo/AppendTag blindly.

i.e:
(if caller passes ctx)
```
logger.GetReqInfo(ctx).AppendTags("endpoint", endpoint.String())
logger.LogIf(ctx, err)
```
or
```
(if caller does not pass the ctx)
ctx := context.Background()
logger.GetReqInfo(ctx).AppendTags("endpoint", endpoint.String())
logger.LogIf(ctx, err)
```

Reply:
Fixed